presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/CteProjectionAndPredicatePushDown.java

```diff
@@ -370,14 +370,21 @@ private PlanNode addFilter(PlanNode node, List<RowExpression> predicates)
                 return node;
             }
 
-            RowExpression predicate;
+            RowExpression resultPredicate;
             if (predicates.size() == 1) {
-                predicate = predicates.get(0);
+                resultPredicate = predicates.get(0);
             }
             else {
-                predicate = new SpecialFormExpression(OR, BOOLEAN, predicates);
+                resultPredicate = predicates.get(0);
+                for (int i = 1; i < predicates.size(); i++) {
+                    resultPredicate = new SpecialFormExpression(
+                            SpecialFormExpression.Form.OR,
+                            BOOLEAN,
+                            resultPredicate, predicates.get(i));
+                }
             }
-            return new FilterNode(node.getSourceLocation(), idAllocator.getNextId(), node, predicate);
+            resultPredicate = SimplifyRowExpressions.rewrite(resultPredicate, metadata, session.toConnectorSession());
```

Comment:
Simplifying the **or** conditions is essential and I observed a case in prod where it helped these filters pushed further down in predicatePushdown rule

Reply:
Curious what does this simplify do? Is it to rebalance multiple OR condition?

Reply:
Yeah looks like it breaks down the multiple ors into efficient cnf and dnf forms. In the predicate pushdown optimizer, the current visitJoin method appears to only decompose projects when they are pre broken by this optimizer. 
I've noticed instances where 'or' conditions were not pushed down beneath joins.